opentracing/README.md

```diff
@@ -0,0 +1,121 @@
+# DCP RFC
+
+For background:
+- [Main DCP doc](https://paper.dropbox.com/doc/Distributed-Context-Propagation-RGvlvD1NFKYmrJG9vGCES)
+- [DCP "Layers" doc](https://paper.dropbox.com/doc/DCP-Layers-and-Semantics-X1Tm1MSiBJECBkWYQKS2I)
+
+## API overview for those adding instrumentation
+
+Everyday consumers of this `opentracing` package really only need to worry
+about a couple of key abstractions: the `StartSpan` function, the `Span`
+interface, and golang's `context.Context` idiom. Here are code snippets
+demonstrating some important use cases.
+
+A note about `context.Context`: the opentracing API here encourages (but does
+not require) the use of golang-team's `context.Context` scheme. Whether we like
+it or not (this author is on the fence), `context.Context` is here to stay,
+goroutine-local-storage bedamned. When in Rome, do as the Romans do (etc).
+
+#### Singleton initialization
+
+The simplest starting point is `./global.go`. As early as possible, call
+
+    import ".../opentracing"
+    
+    func main() {
+        tracingRecorder := ... // tracing impl specific
+        tracingContextIDSource := ... // tracing impl specific
+        opentracing.InitGlobalTracer(tracingRecorder, tracingContextIDSource)
+        ...
+    }
+
+##### Note: the singletons are optional
+
+If global singletons make you sad, use `opentracing.NewStandardTracer(...)`
+directly and manage ownership of the `opentracing.OpenTracer` explicitly.
+
+#### Creating a Span given an existing Golang `context.Context`
+
+    func xyz(ctx context.Context, ...) {
+        ...
+        sp, ctx := opentracing.StartSpan("span_name", ctx)
```

Comment:
Great question about the service name -- it's quasi-supported via the `SetTags()` at the `Recorder` (rather than `Span`) level, but I think it should be promoted from a mere optional tag in that interface. I also think it appropriate to rename `Recorder` since it's not quite the right thing, semantically... maybe more like `ProcessRecorder`?

Nit/thought: I hesitate to call it "service name" since there are also non-service components of distributed systems (mobile clients, for instance). Maybe "component name"???